CONTRIBUTING.md

```diff
@@ -1,112 +1,80 @@
-4. Review the MeiliSearch [workflow](#workflow) and [development](#development).
-5. Make your changes.
-6. [Submit the branch as a pull request](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork) to the main MeiliSearch
-   repo. A MeiliSearch team member should comment and/or review your pull request
-   with a few days. Although, depending on the circumstances, it may take
-   longer.
-
-## Change Control
-
-### Git Branches
+4. Review the [Development Workflow](#development-workflow) section that describes the steps to maintain the repository.
+5. Make your changes on your branch.
+6. [Submit the branch as a pull request](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request-from-a-fork) pointing to the `main` branch of the main MeiliSearch repository. A maintainer should comment and/or review your Pull Request within a few days. Although depending on the circumstances, it may take longer.r.
 
-_All_ changes must be made in a branch and submitted as [pull requests](#pull-requests).
-MeiliSearch does not adopt any type of branch naming style, but please use something
-descriptive of your changes.
+## Development Workflow
 
-### Git Commits
+### Setup and run MeiliSearch
 
-#### Style
+```bash
+cargo run
+```
 
-Please ensure your commits are small and focused; they should tell a story of
-your change. This helps reviewers to follow your changes, especially for more
-complex changes.
+We recommend using the `--release` flag to test the full performance of MeiliSearch.
 
-Familiarise yourself with [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/).
+### Test
 
-### Github Pull Requests
+```bash
+cargo test
+```
 
-Once your changes are ready you must submit your branch as a pull request.
+If you get a "Too many open size" error you might want to increase the open file limit using this command:
 
-#### Reviews & Approvals
+```bash
+ulimit -Sn 3000
+```
 
-All pull requests must be reviewed and approved by at least one MeiliSearch team
-member.
+## Git Guidelines
 
-#### Merge Style
-
-All pull requests are squashed and merged. We generally discourage large pull
-requests that are over 300-500 lines of diff. If you would like to propose
-a change that is larger we suggest coming onto our chat channel and
-discuss it with one of our engineers. This way we can talk through the
-solution and discuss if a change that large is even needed! This overall
-will produce a quicker response to the change and likely produce code that
-aligns better with our process.
-
-## Development
+### Git Branches
 
-### Setup
+All changes must be made in a branch and submitted as PR.
 
-See the [MeiliSearch Docs](https://docs.meilisearch.com/reference/features/installation.html) for how to set up a development environment.
+We do not enforce any branch naming style, but please use something descriptive of your changes.
```

Comment:
```suggestion
We do not enforce any branch naming style, but please use something that describes your changes well.
```

Reply:
I took this sentence from the integration team CONTRIBUTING.md and this sentence was previously reviewed by a tech writer, I would rather let it as it is 🙂